vignettes/analyzing-load-test-logs.Rmd

```diff
@@ -12,31 +12,268 @@ vignette: >
+
+As an example, the basic [`01_hello` gallery shiny app](https://gallery.shinyapps.io/001-hello/) loads the homepage, loads more than 20 supporting CSS and JavaScript files, and then begins WebSocket communication with the supporting Shiny server.
+
+
+# Report Output
+
+The data provided by `shinyloadtest::tidy_loadtest` can be used to directly create a stand alone HTML report.  This report contains six major sections of diagnostic information.
+
+```{r, include = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  `1 user` = "../test_sessions/demo1",
+  `4 users` = "../test_sessions/demo4",
+  `16 users` = "../test_sessions/demo16"
+)
+# ignore <- shinyloadtest::make_report(df, "report.html", verbose = FALSE, open_browser = FALSE)
+```
+```{r, eval = FALSE}
+df <- shinyloadtest::tidy_loadtest(
+  "1 user" = "./demo1",
+  "4 users" = "./demo4",
+  "16 users" = "./demo16"
+)
+shinyloadtest::make_report(df, "report.html")
+```
+
+## Sessions
+
+The `Sessions` tab of the shinyloadtest report displays the each simulated session performed by each user.  Each run displays their respective [gantt chart](https://en.wikipedia.org/wiki/Gantt_chart) by clicking on the named buttons near the top of the webpage.
+
+```{r, echo = FALSE, fig.height = 3}
+suppressPackageStartupMessages(library(dplyr, quiet = TRUE))
+suppressPackageStartupMessages(library(shinyloadtest, quiet = TRUE))
+df %>%
+  filter(run == "4 users") %>%
+  plot_gantt()
+```
+
+Sessions that are desaturated in color are sessions that have started before (warmup) or ended after (cooldown) the maintenance period.  The maintenance period is marked with vertical dotted lines spanning every simulated user.
+
+In the example above, four users were stagger started and simulated for 10 minutes. The majority of the time is spent calculating new values for the outputs in the recording.  The session gantt chart below displays the same recording but with sixteen users.  
+
+```{r, echo = FALSE, fig.height = 5}
+df %>%
+  filter(run == "16 users") %>%
+  plot_gantt()
+```
+
+Both runs were executed on a single processor.  With very high resource contention, a larger proportion of time is spent calculating in the `16 users` run.
+
+As more users are placed in an application, the average execution time might go up.  The `4 users` case had 'Simulated User # 0' start almost immediately after mantenance mode began.  However, the `16 users` run should have had a larger stagger start than what was performed.  Most of the other users did not start a new session until 40-50 seconds after 'Simulated User #16' began.  This is not uniform server load behavior.  
+
+It is recommended to uniformly stagger start the user sessions. This can be guestimated by:
```

Comment:
Recent versions of `shinycannon` now automatically stagger start times based on the recording length and number of simulated users, so I think you can just mention that behavior instead of describing how to do it.

Reply:
I agree, you could nix this paragraph and replace the graph above with a graph from a recording with appropriate stagger. I think you want to focus in on the difference in compute time between the two runs, not conflate that with an incorrect stagger.